packages/react-server/src/metrics/test/metric.test.tsx

```diff
@@ -0,0 +1,46 @@
+import {Server} from 'http';
+import React from 'react';
+import request from 'supertest';
+import getPort from 'get-port';
+
+import {metricsMiddleware} from '../metrics';
+import {createServer} from '../../server';
+import {mockMiddleware} from '../../test/utilities';
+
+jest.mock('@shopify/sewing-kit-koa', () => ({
+  middleware: jest.fn(() => mockMiddleware),
+  getAssets() {
+    return {
+      styles: () => Promise.resolve([]),
+      scripts: () => Promise.resolve([]),
+    };
+  },
+}));
+
+describe('createServer()', () => {
+  let server: Server;
+  let port: number;
+  const ip = 'http://localhost';
+
+  beforeEach(async () => {
+    port = await getPort();
+
+    server = await createServer({
+      port,
+      ip,
+      render: () => <div>Mock app</div>,
+    });
+  });
+
+  afterEach(() => {
+    server.close();
+  });
+
+  it('responds with a X-React-Server-Request-Time header', async () => {
```

Comment:
Not sure how useful this is, open to 🔥 if others think it's useless. 

I was going to add another test where the server would contain a `~50ms` delay in the request and the test would assert that the header's value is a reasonable value given the delay. That got a bit gross and didn't seem super useful either.

Reply:
it's pretty medium, but I think overall I'd keep the test /shrug